spec/lib/fastly_nsq/message_queue/producer_spec.rb

```diff
@@ -1,78 +1,51 @@
 require 'spec_helper'
 
 RSpec.describe MessageQueue::Producer do
-  describe '#connection' do
-    describe 'when using the real queue', fake_queue: false do
-      it 'returns an instance of the queue producer' do
-        allow(Nsq::Producer).to receive(:new)
-        topic = 'death_star'
+  let(:topic)    { 'death_star' }
+  let(:producer) { MessageQueue::Producer.new(topic: topic) }
+  let(:backend)  { double 'Producer' }
 
-        MessageQueue::Producer.new(topic: topic).connection
+  describe 'when the ENV is set incorrectly' do
+    it 'raises with a helpful error' do
+      allow(ENV).to receive(:[]).with('FAKE_QUEUE').and_return('taco')
 
-        expect(Nsq::Producer).to have_received(:new).
-          with(
-            nsqd: ENV.fetch('NSQD_TCP_ADDRESS'),
-            topic: topic,
-            ssl_context: nil,
-          ).at_least(:once)
-      end
+      expect { producer.terminate }.to raise_error(InvalidParameterError)
     end
+  end
 
-    describe 'when using the fake queue', fake_queue: true do
-      it 'returns an instance of the queue producer' do
-        allow(FakeMessageQueue::Producer).to receive(:new)
-        topic = 'death_star'
-
-        MessageQueue::Producer.new(topic: topic).connection
-
-        expect(FakeMessageQueue::Producer).to have_received(:new).
-          with(
-            nsqd: ENV.fetch('NSQD_TCP_ADDRESS'),
-            topic: topic,
-            ssl_context: nil,
-          ).at_least(:once)
+  describe 'when connector connects to a backend Producer' do
+    let(:producer) do
+      MessageQueue::Producer.new topic: topic do
+        backend
       end
     end
 
-    describe 'when the ENV is set incorrectly' do
-      it 'raises with a helpful error' do
-        allow(ENV).to receive(:[]).with('FAKE_QUEUE').and_return('taco')
-        topic = 'death_star'
-
-        producer = MessageQueue::Producer.new(topic: topic)
+    it 'forwards #write' do
+      expect(backend).to receive(:write).with("it's a message")
+      producer.write "it's a message"
+    end
 
-        expect { producer.connection }.to raise_error(InvalidParameterError)
-      end
+    it 'forwards #terminate' do
+      expect(backend).to receive(:terminate)
+      producer.terminate
     end
   end
 
-  describe '#terminate' do
-    describe 'when using the real queue', fake_queue: false do
-      it 'closes the connection' do
-        producer = double('Producer', connection: nil, terminate: nil)
-        allow(Nsq::Producer).to receive(:new).and_return(producer)
-        topic = 'death_star'
-
-        live_producer = MessageQueue::Producer.new(topic: topic)
-        live_producer.connection
-        live_producer.terminate
-
-        expect(producer).to have_received(:terminate)
+  describe 'using the default connector' do
+    module TestStrategy
+      module Producer
+        def self.new(*_); end
```

Comment:
I believe it's just `(*args)` but RuboCop wants you to know it's unused, hence the underscore.